integrationTests/testInitializer.go

```diff
@@ -134,6 +134,24 @@ func CreateMessengerWithKadDht(ctx context.Context, initialAddr string) p2p.Mess
 	return libP2PMes
 }
 
+// CreateMessengerWithKadDhtAndProtocolID creates a new libp2p messenger with kad-dht peer discovery and peer ID
+func CreateMessengerWithKadDhtAndProtocolID(ctx context.Context, initialAddr string, randezVous string) p2p.Messenger {
+	p2pConfig := createP2PConfig([]string{initialAddr})
+	p2pConfig.KadDhtPeerDiscovery.RandezVous = randezVous
+	arg := libp2p.ArgsNetworkMessenger{
+		Context:       ctx,
+		ListenAddress: libp2p.ListenLocalhostAddrWithIp4AndTcp,
+		P2pConfig:     p2pConfig,
+	}
+
+	libP2PMes, err := libp2p.NewNetworkMessenger(arg)
+	if err != nil {
+		fmt.Println(err.Error())
```

Comment:
no need for this error - you would return a nil most probably and create panic somewhere else.

Reply:
Agree, but with this print you can easily trace back the problem and fix it in no time.